### ISSUE TITLE
Not working on Moodle 3.9.4+ (Build: 20210122)

### ISSUE THREAD

**Issue by mjthompson** [catalyst/moodle-tool_broadcast#17]
Hi Matt

I have been unable to get this plugin working on the above-mentioned build. I have installed the plugin, set up a broadcast, but it does not display to users.

- Turning on debugging did not reveal anything. 
- No errors reported in browser console.
- Unit tests passed ok.
- I thought it might be theme-related, so I changed back to the default (Boost), but to no avail. 
- Cleared all caches.

Is there a compatibility issue with this Moodle build? If there's any information I can supply let me know.

Thanks
(another) Matt

**Comment by mnods** [comment]
Hi Matt, on 3.9.7 and i'm getting similar issues. 

Also some people got a "Failed to create broadcast modal" ajax type modal error - sorry don't have any debugging specifics but could try to replicate if needed. It's a little erratic - seems to appear for me as admin in one environment but getting nothing at all in another (same set up - mdl 3.9.7 and latest commit of broadcast.)

would be great if this could be made to work in 3.9 as it's potentially very useful 

cheers

**Comment by mdjnelson** [comment]
I can replicate this. The webservice is not returning any data for the user but did for me as the admin when I just created the broadcast.

**Comment by katcher** [comment]
Hello Matt, I installed on our dev quiz site and tried the plugin - from master.  It worked as long as I did not click on Logged In user.    I saw the following error in dev tools:

[
	{
		"error": true,
		"exception": {
			"message": "Invalid user",
			"errorcode": "invaliduser",
			"link": "https://moodle-dev-quiz.xxxx.ca/",
			"moreinfourl": "http://docs.moodle.org/39/en/error/moodle/invaliduser"
		}
	}
]

From what I understood, I set up a broadcast site wide for all Logged in users -> error
When I just created a course wide broadcast without requiring logged in user, the message was displayed.

We are using moodle 3.9.9+

**Comment by mjthompson** [comment]
Sorry Matt for the delay, I've been unable to test this again on our dev
environment. I will get back to you on the next couple of days. Perhaps
there is something unique on our end causing the issue.

**Comment by cameron1729** [comment]
Hi all,

I'm taking over working on this. I've tested latest the master branch on Moodle 3.9.9+ (commit 089f5bb0) and don't see any errors.

If you are still experiencing problems can you please provide detailed, step-by-step replication instructions.

Cheers,

Cam

**Comment by mjthompson** [comment]
Hi @cameron1729 

I don't think the reproduction steps will be too helpful. 
1. git clone https://github.com/catalyst/moodle-tool_broadcast admin/tool/broadcast && php admin/cli/upgrade.php
2. Create a broadcast
3. Go to the dashboard as admin user and the broadcast does not appear.

This is something environment-specific. I installed Moodle 3.9 in a Docker container and the plugin fine.

If I manually make a GET request to /lib/ajax/service.php?sesskey=valid&info=tool_broadcast_get_broadcasts I get the following:

`{"error":"Coding error detected, it must be fixed by a programmer: Invalid json in request: Syntax error","errorcode":"codingerror","stacktrace":"* line 63 of \/lib\/ajax\/service.php: coding_exception thrown\n","debuginfo":"\nError code: codingerror","reproductionlink":"https:\/\/website.redacted.example.com\/"}`

While on the Docker container, the same GET request (valid session key substituted) returns:

`[{"error":false,"data":"{\"1\":{\"id\":\"1\",\"title\":\"Test\",\"body\":\"<p dir=\\\"ltr\\\" style=\\\"text-align: left;\\\">Test<br><\\\/p>\",\"loggedin\":\"0\",\"timestart\":\"1630759020\",\"mode\":\"1\"}}"}]`

Since this is working on a stock install, I'm happy to close the issue, but it's still not working for me I'm afraid.

**Comment by mjthompson** [comment]
@cameron1729:

I tested this on a Docker container with 3.9 and it worked. So, out of desperation after debugging that was going nowhere, I switched our test environment back to the default theme and the plugin is working as expected. This must be an issue with our theme. 

What is strange I *did* switch back to the default theme before filing this issue and the plugin was also not working. Perhaps the conjunction of two issues caused a quagmire - who knows. 

Thanks all for your assistance @cameron1729 & @mdjnelson & @katcher
